resolver/resolver.go

```diff
@@ -38,7 +39,8 @@ func New(version string, config records.Config) *Resolver {
 		version: version,
 		config:  config,
 		rs:      &records.RecordGenerator{},
-		rng:     rand.New(rand.NewSource(time.Now().UnixNano())),
+		// See: https://github.com/golang/go/issues/3611
+		rng:     rand.New(&lockedSource{src: rand.NewSource(time.Now().UnixNano())}),
```

Comment:
Reading through https://github.com/golang/go/issues/3611 clarified what you are trying to do. With some explaining text in the commit message it would be fine. Placing a random link somewhere in the code is not nice.